### PR TITLE
NAIP test reconstructions

### PR DESCRIPTION
**Pull request by qhuang62**
Add NAIP RGB reconstruction pipeline
- Download script for paired NAIP (RGB) + CHM (LiDAR elevation) data (hard coded for zone15 for test)
- Parser to create (x,y,z,t,r,g,b) training tensors from GeoTIFFs
- Training script for Earth4D + MLP head RGB reconstruction
- Visualization script for 4-panel comparison figure
- Complete pipeline documentation in NAIP_PIPELINE.md

Results: Learned probing achieves 18% validation loss improvement over baseline (0.00171 vs 0.00209 MSE) on 7.3M point Houston dataset (5.8M train, 1.5M validation, 80/20 split).

Also includes bug fix for earth4d.py .to() method for collision tracking; and the visualization script for the collision rate heatmaps across synthetic simulations, which were not include in previous PR.

### CODE REVIEW

**Comment by qhuang62** [comment]
I was just reviewing the code closely and found that the training script uses random point splitting (random_split) while the visualization assumes chip-level splits -- meaning the “validation chip” may include training data. Will need to switch to a chip-level split for consistency.

**Comment by legel** [comment]
Got it @qhuang62, thanks for the heads up on that. Feel free to share an updated figure with chip-level test.

Great work.

**Comment by legel** [comment]
Hey @qhuang62 I was looking to dive in here, wanted to see where things are at for you, if I should hold off.

Besides the train/test split that maybe is based on chip, e.g., one thing that I saw will be good to knock out is this:

```
The team lead requested 5 specific target locations:
1. ASU campus (Tempe, AZ) - 33.42°N, 111.93°W - **UTM Zone 12**
2. Boulder, CO (Flatirons) - 40.01°N, 105.27°W - **UTM Zone 13**
3. NYC downtown (High Line) - 40.75°N, 74.00°W - **UTM Zone 18**
4. Stanford University - 37.43°N, 122.17°W - **UTM Zone 10**
5. South Beach Miami - 25.76°N, 80.13°W - **UTM Zone 17**

**WE DID NOT DOWNLOAD THESE LOCATIONS.** We downloaded UTM Zone 15 (Houston/Texas area) which does not contain any of the 5 target regions. The `download_naip_all.py` script is hardcoded to Zone 15 and would need modification to download the correct zones (10, 12, 13, 17, 18).
```

Technically I'm less interested in those 5 locations, and more interested in a system that is deft-enough to download any NAIP + 3DEP pair quickly given any (latitude, longitude). It does appear that e.g. a Python library for UTM <-> GPS can help.
